### PR TITLE
feat: change logic of resource name length check

### PR DESCRIPTION
**Pull request by Pythonberg1997**
### Description

This pr is to change the resource name length limit from bytes length to character length

### Changes

Notable changes: 
* change the resource name length limit from bytes length to character length

### Potential Impacts
* related check in sdk

### CODE REVIEW

File: x/storage/types/crosschain.go [hunk 1]
```diff
@@ -562,7 +562,7 @@ func (p CreateBucketSynPackage) ValidateBasic() error {
 		ChargedReadQuota: p.ChargedReadQuota,
 	}
 
-	return msg.ValidateBasic()
+	return msg.ValidateRuntime(ctx)
```
Comment (zjubfd) [inline x/storage/types/crosschain.go:565]: you can do `ValidateBasic` and `ValidateRuntime` independently

File: x/storage/types/message.go [hunk 2]
```diff
@@ -180,10 +181,24 @@ func (msg *MsgCreateBucket) ValidateBasic() error {
 		return errors.Wrapf(ErrInvalidVisibility, "Unspecified visibility is not allowed.")
 	}
 
-	err = s3util.CheckValidBucketName(msg.BucketName)
+	return nil
+}
+
+func (msg *MsgCreateBucket) ValidateRuntime(ctx sdk.Context) error {
+	err := msg.ValidateBasic()
```
Comment (zjubfd) [inline x/storage/types/message.go:188]: here need change

File: x/storage/types/message.go [hunk 3]
```diff
@@ -227,7 +242,20 @@ func (msg *MsgDeleteBucket) ValidateBasic() error {
 		return errors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
 	}
 
-	err = s3util.CheckValidBucketName(msg.BucketName)
+	return nil
+}
+
+func (msg *MsgDeleteBucket) ValidateRuntime(ctx sdk.Context) error {
+	err := msg.ValidateBasic()
```
Comment (zjubfd) [inline x/storage/types/message.go:249]: so is here.

File: x/storage/types/message.go [hunk 4]
```diff
@@ -293,6 +317,24 @@ func (msg *MsgUpdateBucketInfo) ValidateBasic() error {
 	return nil
 }
 
+func (msg *MsgUpdateBucketInfo) ValidateRuntime(ctx sdk.Context) error {
+	err := msg.ValidateBasic()
```
Comment (zjubfd) [inline x/storage/types/message.go:321]: so is here

File: x/challenge/types/message_submit.go [hunk 5]
```diff
@@ -55,12 +56,29 @@ func (msg *MsgSubmit) ValidateBasic() error {
 		return errors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid sp operator address (%s)", err)
 	}
 
-	if err = s3util.CheckValidBucketName(msg.BucketName); err != nil {
+	return nil
+}
+
+func (msg *MsgSubmit) ValidateRuntime(ctx sdk.Context) error {
+	err := msg.ValidateBasic()
```
[the remark this answers is not in the document]
Comment (Pythonberg1997) [reply x/challenge/types/message_submit.go:63]: Fixed

File: x/storage/types/crosschain.go (hunk 1 above)
Comment (Pythonberg1997) [reply x/storage/types/crosschain.go:565]: Done

File: x/storage/types/message.go (hunk 2 above)
Comment (Pythonberg1997) [reply x/storage/types/message.go:188]: Done

File: x/storage/types/message.go (hunk 3 above)
Comment (Pythonberg1997) [reply x/storage/types/message.go:249]: Done

File: x/storage/types/message.go (hunk 4 above)
Comment (Pythonberg1997) [reply x/storage/types/message.go:321]: Done

File: go.mod [hunk 6]
```diff
@@ -180,7 +180,7 @@ replace (
 	github.com/cometbft/cometbft => github.com/bnb-chain/greenfield-cometbft v1.1.0
 	github.com/cometbft/cometbft-db => github.com/bnb-chain/greenfield-cometbft-db v0.8.1-alpha.1
 	github.com/confio/ics23/go => github.com/cosmos/cosmos-sdk/ics23/go v0.8.0
-	github.com/cosmos/cosmos-sdk => github.com/bnb-chain/greenfield-cosmos-sdk v1.2.2-0.20240102032738-3661d43faf8f
+	github.com/cosmos/cosmos-sdk => github.com/Pythonberg1997/greenfield-cosmos-sdk v0.0.0-20240112053628-9e3d8de9d89e
```
Comment (zjubfd) [inline go.mod:183]: Fix the denpendency

**Review by yutianwu — APPROVED** [review]

**Review by zjubfd — APPROVED** [review]

**Pull request merged**